src/main/java/Data/UserManager.java

```diff
@@ -0,0 +1,61 @@
+package Data;
+
+import Library.Book;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class UserManager {
+
+    private final String filePath = "C:\\Users\\david\\IdeaProjects\\LibraryProject\\Online-Library\\src\\main\\resources\\users.txt";
+
+    private List<User> userList;
+
+    public UserManager() {
+        this.userList = readAllUsers();
+    }
+
+    public User registerUser(String username, String password, String email) {
+        String userID = UUID.randomUUID().toString();
+        User newUser = new User(userID, username, password, email);
+
+        try {
+            FileWriter writer = new FileWriter("C:\\Users\\david\\IdeaProjects\\LibraryProject\\Online-Library\\src\\main\\resources\\users.txt", true);
+            writer.write(newUser + "\n");
+            writer.close();
```

Comment:
you can move writer.close() line to finally clause, but even better try to use try-with-resources like in Book Handler

Reply:
done